lib/api/authentication/web_authentication.py

```diff
@@ -35,9 +43,13 @@ def authenticate(self, request):
         self.enforce_csrf(request)
         
         if 'Authorization' in request.headers:
-            token = request.headers['Authorization'].split()[1]
-            payload = jwt.decode(token, settings.SECRET_KEY) #TODO check if valid and handle bad signature
-            web_token = WebToken(payload['uid'], payload['course'])
+            try:
+                scheme, token = request.headers['Authorization'].split()
+                if scheme != 'Bearer':
+                    raise ValueError
+            except ValueError:
+                return (user, None)
+            web_token = WebToken.from_jwt(token)
```

Comment:
You should capture the verification failure, which can be raised by the `jwt.decode`.